src/WARP.XrmSolutionAssistant/WARP.XrmSolutionAssistant.Core/SolutionFlowConnectionMapper.cs

```diff
@@ -0,0 +1,247 @@
+
+        private readonly string solutionRootDirectory;
+
+        private readonly Dictionary<string, string> replacements;
+
+        private readonly List<string> unmappedApis;
+
+        private readonly List<string> customizationConnectionsToDelete;
+
+        private SettingsWrapper settings;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="SolutionFlowConnectionMapper"/> class.
+        /// </summary>
+        /// <param name="solutionRootDirectory">Path to the directory containing the extracted solution.</param>
+        public SolutionFlowConnectionMapper(string solutionRootDirectory)
+        {
+            this.solutionRootDirectory = solutionRootDirectory;
+            this.replacements = new Dictionary<string, string>();
+            this.unmappedApis = new List<string>();
+            this.customizationConnectionsToDelete = new List<string>();
+            this.settings = Workers.Settings.GetSettings();
+        }
+
+        /// <summary>
+        /// Executes the logic to map the Flow connection references to the values defined in settings.json.
+        /// </summary>
+        public void Execute()
+        {
+            Logger.Info("Solution Root Directory: {0}", this.solutionRootDirectory);
+
+            if (!Directory.Exists(this.solutionRootDirectory))
+            {
+                Logger.Fatal("The given solution root directory does not exist or is not available. Exiting.");
+                return;
+            }
+
+            try
+            {
+                var otherDir = Path.Combine(this.solutionRootDirectory, "Other");
+                var customizationsXmlPath = Path.Combine(otherDir, "customizations.xml");
+
+                Logger.Info("Getting connection references from customizations.xml.");
+                var customizationsConnReferences = this.GetConnectionReferencesFromCustomizations(customizationsXmlPath);
+                Logger.Info($"{customizationsConnReferences.Count} connection references retrieved from customizations.xml.");
+
+                this.ParseCustomizationReferences(customizationsConnReferences);
+                if (this.unmappedApis.Any())
+                {
+                    var unmappedApisString = $"The following Flow apis do not have mappings; {string.Join(",", this.unmappedApis.Distinct())}";
+                    if (this.settings.FlowConnectionMapper.ThrowExceptionOnUnmappedConnections)
+                    {
+                        Logger.Error(unmappedApisString);
+                        throw new Exception(unmappedApisString);
+                    }
+
+                    Logger.Warn(unmappedApisString);
+                }
+
+                this.ProcessCustomizationsFile(customizationsXmlPath);
+                this.ProcessFlowJsons();
+            }
+            catch (Exception ex)
+            {
+                Logger.Fatal(ex, "Unexpected error: {0}", ex);
+                throw ex;
```

Comment:
Should be using just 'throw' here to avoid claring the stack trace.

Reply:
Fixed.